electron/execution.js

```diff
@@ -126,52 +126,70 @@ async function recordJourneys(data, browserWindow) {
   }
 }
 
-async function onTest(data) {
-  const result = {
-    succeeded: 0,
-    failed: 0,
-    skipped: 0,
-    journey: {},
-  };
-
+async function onTest(data, browserWindow) {
   const parseOrSkip = chunk => {
-    try {
-      return JSON.parse(chunk);
-    } catch (_) {
-      return {};
-    }
+    // at times stdout ships multiple steps in one chunk, broken by newline,
+    // so here we split on the newline
+    return chunk.split("\n").map(subChunk => {
+      try {
+        return JSON.parse(subChunk);
+      } catch (_) {
+        return {};
+      }
+    });
   };
-  const constructResult = chunk => {
-    const parsed = parseOrSkip(chunk);
+
+  // returns TestEvent interface defined in common/types.ts
```

Comment:
At some point we should probably transpile this file before running it so that we can have types in it.

In the meantime, I think it could be nice to transform this comment into a JSDoc style block. 

That's no big deal though, thanks for specifying the return type here!

Reply:
I think [that's a great idea](https://github.com/elastic/synthetics-recorder/issues/141) 😉

Reply:
There is some discussion on that issue still though so we should make sure all are on the same page before committing to anything.

Reply:
I actually considered the JSDoc style :D It seemed to be a bit overkill as the `TestEvent` interface is union type of three which is a bit verbose to cover it in JSDoc, as well as the number of the TestEvent variation, will be growing. If we decided not to convert it to TS any time soon, I will def add JSDoc block 👍